app/controllers/region_admin/donors_controller.rb

```diff
@@ -0,0 +1,36 @@
+class RegionAdmin::DonorsController < ApplicationController
+  before_filter :authenticate_volunteer!
+  before_filter :authorize_region_admin!
+
+  def index
+    @locations = regional_donors
+    @regions   = available_regions
+  end
+
+  private
+
+  def regional_donors
+    Location.active.donors.regional(region_ids)
+  end
+
+  def region_ids
+    current_volunteer.region_ids
+  end
+
+  def available_regions
+    if current_volunteer.super_admin?
+      Region.all
+    else
+      current_volunteer.assignments.collect{ |a| a.admin ? a.region : nil }.compact
+    end
+  end
+
+  def authorize_region_admin!
+    return if region_admin?
+    redirect_to root_url, alert: "Unauthorized"
```

Comment:
Could we do one line: `redirect_to root_url, alert: "Unauthorized" unless region_admin?` ?